ts/components/messages/MessageDetailComponent.tsx

```diff
@@ -121,24 +121,26 @@ export default class MessageDetailComponent extends React.PureComponent<Props> {
       <View style={styles.mainWrapper}>
         <View style={styles.headerContainer}>
           {/* Service */}
-          {service && (
-            <Grid style={styles.serviceContainer}>
-              <Col>
-                <H4>{service.organization_name}</H4>
-                <H6 link={true} onPress={onServiceLinkPress}>
-                  {service.service_name}
-                </H6>
-              </Col>
-              <Col style={styles.serviceCol}>
-                <TouchableOpacity onPress={onServiceLinkPress}>
-                  <MultiImage
-                    style={styles.serviceMultiImage}
-                    source={logosForService(service)}
-                  />
-                </TouchableOpacity>
-              </Col>
-            </Grid>
-          )}
+          {service &&
+            service.service_id &&
+            pot.isSome(potService) && (
+              <Grid style={styles.serviceContainer}>
+                <Col>
+                  <H4>{service.organization_name}</H4>
+                  <H6 link={true} onPress={onServiceLinkPress}>
+                    {service.service_name}
```

Comment:
Adding **_pot.isSome(potService)_** nothing will be show if service is unknown (_pot.none_)
Could be a better approach if we add **_service_name_** here
https://github.com/teamdigitale/io-app/blob/48a27ac1cb5a0675935770cd1fab961bf164af61/ts/components/messages/MessageDetailComponent.tsx#L105 
and then render the logo only if **_service_id_** is defined?

Reply:
From the service definition I read  `service_id` is available only if `potService` is `some`. Is it a wrong reading of the service definition?

Reply:
Yes of course, **_service_id_** is available only if pot is some.
Anyway we should show info about the service is unknown.

**This is how a full loaded message looks like**
<img src="https://user-images.githubusercontent.com/822471/64028980-fbd81300-cb43-11e9-8909-5d5156acb36d.png" height="500"/>

With this approach no info about service is shown
<img src="https://user-images.githubusercontent.com/822471/64028513-13fb6280-cb43-11e9-9d9d-9a5e13fd3e9f.png" height="500"/>

Changing the code in this way

```typescript
const service =
      potService !== undefined
        ? pot.isNone(potService)
          ? ({
              organization_name: I18n.t("messages.errorLoading.senderService"),
              department_name: I18n.t("messages.errorLoading.senderInfo"),
              service_name: ""
            } as ServicePublic)
          : pot.toUndefined(potService)
        : undefined;

    const payment =
      message.content.payment_data !== undefined &&
      service !== undefined &&
      service.organization_fiscal_code
        ? paymentsByRptId[
            `${service.organization_fiscal_code}${
              message.content.payment_data.notice_number
            }`
          ]
        : undefined;

    return (
      <View style={styles.mainWrapper}>
        <View style={styles.headerContainer}>
          {/* Service */}
          {service && (
            <Grid style={styles.serviceContainer}>
              <Col>
                <H4>{service.organization_name}</H4>
                <H6 link={true} onPress={onServiceLinkPress}>
                  {service.service_name}
                </H6>
              </Col>
              {service.service_id && (
                <Col style={styles.serviceCol}>
                  <TouchableOpacity onPress={onServiceLinkPress}>
                    <MultiImage
                      style={styles.serviceMultiImage}
                      source={logosForService(service)}
                    />
                  </TouchableOpacity>
                </Col>
              )}
            </Grid>
          )}
```
service details will be shown
<img src="https://user-images.githubusercontent.com/822471/64028841-ad2a7900-cb43-11e9-81fd-a40bcb34c034.png" height="500"/>

I think it's preferred to show a message instead of nothing